### ISSUE TITLE
The problem with selection of the contents inside the brackets.

### ISSUE THREAD

**Issue by johnkind49** [facelessuser/BracketHighlighter#257]
After updating the brackets are searched much better, and it's great. However, there is a problem with selection of the content inside the brackets (sometimes selected parent content). https://youtu.be/U5594bwFG0E

**Comment by facelessuser** [comment]
I know this may seem confusing, but it is expected.  The current implementation of bracketselect.py handles the selection of content and moving the caret/cursor (basically zero width selections).  So if we run a jump to end bracket command it we get this:

` { { tes|t } more text }` -> `{ { test |} more text }`

We run it again, we get :

` { { test |} more text }` -> `{ { test }| more text }`

And again:

` { { test }| more text }` -> `{ { test } more text |}`

And so on. So this is nice functionality.  And I bet you can see what I am using to trigger the logic; I check if the caret is on the inside or outside.  So you can see, even though these brackets highlight on the outside with `bracket_outside_adjacent` enabled , when you invoke the command, we run without "outside adjacent" matching.

```js
// Go to left bracket
    {
        "caption": "BracketHighlighter: Jump to Left Bracket",
        "command": "bh_async_key",
        "args":
        {
            "no_outside_adj": true,  // <----------- No outside adjacent
            "lines" : true,
            "plugin":
            {
                "type": ["__all__"],
                "command": "bh_modules.bracketselect",
                "args": {"select": "left"}
            }
        }
    },
```

Now I know what your thinking, let's split the logic, but what about expanding selections?  With this logic in place, we can do this on running the selection command ('<>` will be highlights).

` { { tes|t } more text }` -> `{ {< test >|} more text }`

again

` { {< test >|} more text }` -> `{ <{ test }>| more text }`

again

` { <{ test }>| more text }` -> `{< { test } more text >|}`

etc.

So this is my intended functionality.  So your caret must be inside the brackets to select the content inside.

You can create your own custom keybinding or command and set `no_outside_adj` to `false`.  If you like that functionality, have at it.  Keep in mind this changes functionality as discussed here in #187.

Bracket plugins only know about the current bracket; not the parent. So the only way to get the parent bracket instead of the current is to ignore outside adjacent matching.  This is fully intentional and is the current limitation of the bracket matching framework.

**Comment by johnkind49** [comment]
Thank you, I understood the logic of "no_outside_adj", not a bug, but a feature.
"no_outside_adj => false" solves my problem.

However, on the menu it seems more logical to make the option "no_outside_adj => false" by default.
Because with the new option "bracket_outside_adjacent => true" is unclear why the text is selected from the outside, rather than inside, without further explanation.
Or why the command "bh_key" moves to the parent bracket, while the children are marked.

P.S. Sorry for my bad English.

**Comment by facelessuser** [comment]
> However, on the menu it seems more logical to make the option "no_outside_adj => false" by default.

Ahh, but you forget that `bracket_outside_adjacent ` is disabled by default.  So by default, no one has this problem, and they get the expanding selections and the ability to jump to parent brackets without any confusion.

`bracket_outside_adjacent ` was only added to appease the masses who were requesting it, but BH was built around it being off by default.  Your issue is one of the small compromises that have to be made to have the outside adjacent feature.

Maybe I can add a command wrapper that for selection commands that allows for users to control the `no_outside_adj` functionality via the settings file.  I think that would be the best approach.

> P.S. Sorry for my bad English.

No problem.

**Comment by johnkind49** [comment]
>Ahh, but you forget that bracket_outside_adjacent is disabled by default. So by default, no one has this problem, and they get the expanding selections and the ability to jump to parent brackets without any confusion.

Indeed, if it were so - the problem would not exist.
But in the default configuration file option "bracket_outside_adjacent => true":
https://github.com/facelessuser/BracketHighlighter/blob/master/bh_core.sublime-settings#L48
That's what is confusing.

**Comment by facelessuser** [comment]
> Indeed, if it were so - the problem would not exist.
But in the default configuration file option "bracket_outside_adjacent => true":
https://github.com/facelessuser/BracketHighlighter/blob/master/bh_core.sublime-settings#L48
That's what is confusing.

Touché.  I didn't realize I had that enabled; it wasn't supposed to be.  #258 should still provide an easy way for a user to adjust the behavior moving forward which is what I think needs to happen.  As long as a the behavior is documented and a way for people to adjust it, I think we should be fine.

**Comment by facelessuser** [comment]
I decided to expose a setting called `ignore_outside_adjacent_in_pugin` in the settings file that can globally control this for a user if in the command `no_outside_adj` is set to `null` (or `None` in Python).  All selection commands will use `null` by default allowing users to just pop into the settings file alter the behavior of the default commands as they see fit.  This will be released in *2.9.0* whenever I get around to that.
